### ISSUE TITLE
ME_CONFIG_BASICAUTH_USERNAME empty but ask for a username/password in basic auth

### ISSUE THREAD

**Issue by fkarakas** [mongo-express/mongo-express#1537]
Steps:

- Set ME_CONFIG_BASICAUTH_USERNAME to empty string; deploy the docker image.

Result:
You get a basic auth authentication login on your browser 

Expected result:
No basic auth

Remarks:
As mentioned in the documentation, setting this username to empty should disable basic auth

```
mongo-express web login name. Sending an empty string will disable basic authentication.
```

Other configuration:
  ME_CONFIG_MONGODB_URL: <mongo_url>
  ME_CONFIG_BASICAUTH_USERNAME: ""
  ME_CONFIG_BASICAUTH_PASSWORD: ""
  ME_CONFIG_OPTIONS_EDITORTHEME: "ambiance"
  ME_CONFIG_OPTIONS_READONLY: "true"
  ME_CONFIG_MONGODB_ENABLE_ADMIN: "false"

**Comment by xTudoS** [comment]
what version docker did you use?
docker from docker hub is out-of-date

**Comment by kpruden** [comment]
I'm seeing this issue too. I'm running mongo-express 1.0.2 from docker hub.

**Comment by fkarakas** [comment]
I am using docker hub what should i use?

**Comment by xTudoS** [comment]
> I'm seeing this issue too. I'm running mongo-express:1.0.2 from docker hub.

> I am using docker hub what should i use?

Build your own docker image, this repository can help: https://github.com/mongo-express/mongo-express-docker

docker-hub and npm package are out-of-date
so any documentation in this repository readme will not work in docker hub version

**Comment by kpruden** [comment]
Are there plans to push an updated image to docker hub?

**Comment by traceroute42** [comment]
Inside pod I found in printenv
`ME_CONFIG_BASICAUTH=true`

setting up in deployment env 
```
        - name: ME_CONFIG_BASICAUTH
          value: "false"
```

disabled basic auth maybe it will help image is image: mongo-express

**Comment by fkarakas** [comment]
well done it seems to work ! but why ?
in the documentation it says that it is default to false
anyway

**Comment by JavertArdo** [comment]
The browser asks for the user and password due to the change requested in `mongo-express-docker` repository [#117](https://github.com/mongo-express/mongo-express-docker/issues/117), delivered in [#118](https://github.com/mongo-express/mongo-express-docker/pull/118). The `ME_CONFIG_BASICAUTH` was hardcoded in Dockerfiles to `true`. As I understand this repository is used to build images for Docker Hub as the `README.md` file from there is being shown on that platform: [Mongo Express](https://hub.docker.com/_/mongo-express).

I am wondering why they decided to force enable it in containers, since it was disabled for almost two years [Disable the default basic authentication #975](https://github.com/mongo-express/mongo-express/pull/975/commits/544b94d6f9f506af293fe8cc0525f344decce131). Even the `README.md` in `mongo-express-docker` do not mentions about enabled authentication by default.

This change is very confusing, since the original `mongo-express` repository mentions in `README.md` that this environment variable is by default `false`.

Another problem might be the `ME_CONFIG_BASICAUTH_USERNAME`. The value is indeed taken from that env, but there is a file `config.default.js` and following part of the code:

```js
  basicAuth: {
    username: getFileEnv(basicAuthUsername) || 'admin',
    password: getFileEnv(basicAuthPassword) || 'pass',
  },
```

It is worth to remember if any of the environment variables `ME_CONFIG_BASICAUTH_USERNAME` or `ME_CONFIG_BASICAUTH_PASSWORD` is not defined, then default value will be respectively `admin` and `pass`.

Setting `ME_CONFIG_BASICAUTH_USERNAME` to empty string won't disable authentication, since [Disable the default basic authentication #975](https://github.com/mongo-express/mongo-express/pull/975/commits/544b94d6f9f506af293fe8cc0525f344decce131) introduced `ME_CONFIG_BASICAUTH` and forgot to update `README.md`.

Before change:
```js
useBasicAuth: getFileEnv(basicAuthUsername) !== '',
```

After change:
```js
useBasicAuth: getBoolean(getFileEnv(basicAuth)),
```

Anyway, I think that `ME_CONFIG_BASICAUTH` name is apart from the naming convention and more suitable could be `ME_CONFIG_BASICAUTH_ENABLED`.

**Comment by xTudoS** [comment]
> The browser asks for the user and password due to the change requested in `mongo-express-docker` repository [#117](https://github.com/mongo-express/mongo-express-docker/issues/117), delivered in [#118](https://github.com/mongo-express/mongo-express-docker/pull/118). The `ME_CONFIG_BASICAUTH` was hardcoded in Dockerfiles to `true`. As I understand this repository is used to build images for Docker Hub as the `README.md` file from there is being shown on that platform: [Mongo Express](https://hub.docker.com/_/mongo-express).
> 
> I am wondering why they decided to force enable it in containers, since it was disabled for almost two years [Disable the default basic authentication #975](https://github.com/mongo-express/mongo-express/pull/975/commits/544b94d6f9f506af293fe8cc0525f344decce131). Even the `README.md` in `mongo-express-docker` do not mentions about enabled authentication by default.
> 
> This change is very confusing, since the original `mongo-express` repository mentions in `README.md` that this environment variable is by default `false`.
> 
> Another problem might be the `ME_CONFIG_BASICAUTH_USERNAME`. The value is indeed taken from that env, but there is a file `config.default.js` and following part of the code:
> 
> ```js
>   basicAuth: {
>     username: getFileEnv(basicAuthUsername) || 'admin',
>     password: getFileEnv(basicAuthPassword) || 'pass',
>   },
> ```
> 
> It is worth to remember if any of the environment variables `ME_CONFIG_BASICAUTH_USERNAME` or `ME_CONFIG_BASICAUTH_PASSWORD` is not defined, then default value will be respectively `admin` and `pass`.
> 
> Setting `ME_CONFIG_BASICAUTH_USERNAME` to empty string won't disable authentication, since [Disable the default basic authentication #975](https://github.com/mongo-express/mongo-express/pull/975/commits/544b94d6f9f506af293fe8cc0525f344decce131) introduced `ME_CONFIG_BASICAUTH` and forgot to update `README.md`.
> 
> Before change:
> 
> ```js
> useBasicAuth: getFileEnv(basicAuthUsername) !== '',
> ```
> 
> After change:
> 
> ```js
> useBasicAuth: getBoolean(getFileEnv(basicAuth)),
> ```
> 
> Anyway, I think that `ME_CONFIG_BASICAUTH` name is apart from the naming convention and more suitable could be `ME_CONFIG_BASICAUTH_ENABLED`.

Thank you for all information. Would you mind to send a PR to improve this infos?

**Comment by JavertArdo** [comment]
Should we consider renaming `ME_CONFIG_BASICAUTH` to `ME_CONFIG_BASICAUTH_ENABLED`? If yes, then I can prepare pull request with change and `README.md` update.

To keep legacy behaviour I suggest to have both environment variables for now, but `ME_CONFIG_BASICAUTH_ENABLED` should have priority over `ME_CONFIG_BASICAUTH`. Additionally `ME_CONFIG_BASICAUTH` can be marked in docs as deprecated.

**Comment by xTudoS** [comment]
@rtritto @shakaran @BlackthornYugen what do you guys think?
I agree with this change, `ME_CONFIG_BASICAUTH_ENABLED `  seems  more specific

**Comment by BlackthornYugen** [comment]
Yeah, that makes sense. I'd prefer the more explicit BASICAUTH_ENABLED as has been suggested.

**Comment by rtritto** [comment]
Closed by #1606
